src/tests/PubSub.test.ts

```diff
@@ -0,0 +1,252 @@
+      isNewSpy = jest.spyOn(PubSub.prototype as any, "isNew");
+      hashDataSpy = jest.spyOn(PubSub.prototype as any, "hashData");
+    });
+
+    afterEach(() => {
+      isNewSpy.mockReset();
+      hashDataSpy.mockReset();
+    });
+
+    afterAll(() => {
+      isNewSpy.mockRestore();
+      hashDataSpy.mockRestore();
+    });
+
+    it("should call isNew", async () => {
+      isNewSpy.mockResolvedValueOnce(false);
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(isNewSpy).toHaveBeenCalledTimes(1);
+      expect(isNewSpy).toHaveBeenCalledWith({ test: "test" }, "test");
+    });
+
+    it("should do nothing if the message isn't new", async () => {
+      isNewSpy.mockResolvedValueOnce(false);
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(isNewSpy).toHaveBeenCalledTimes(1);
+      expect(isNewSpy).toHaveBeenCalledWith({ test: "test" }, "test");
+      expect((pubSub as any).client.publishJSON).not.toHaveBeenCalled();
+    });
+
+    it("should call hashData", async () => {
+      isNewSpy.mockRestore();
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(hashDataSpy).toHaveBeenCalledTimes(1);
+      expect(hashDataSpy).toHaveBeenCalledWith({ test: "test" });
+      isNewSpy = jest.spyOn(PubSub.prototype as any, "isNew");
+    });
+
+    it("should save MessagesSent", async () => {
+      isNewSpy.mockResolvedValueOnce(true);
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(typeorm.getManager).toHaveBeenCalledTimes(1);
+      expect(typeorm.getManager().getRepository).toHaveBeenCalledTimes(1);
+      expect(typeorm.getManager().getRepository).toHaveBeenCalledWith(
+        MessagesSent
+      );
+      expect(
+        typeorm.getManager().getRepository(MessagesSent).save
```

Comment:
it is but to get to the save mock we need to follow the chain. otherwise we would need to create the mocks outside for the es6 class mock and use these inside the mock implementation